pkg/repo/migrations/v3_4.go

```diff
@@ -0,0 +1,52 @@
+package migrations
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	"github.com/bacalhau-project/bacalhau/pkg/repo"
+)
+
+var V3Migration = repo.NewMigration(
+	repo.Version3,
+	repo.Version4,
+	func(r repo.FsRepo) error {
+		repoPath, err := r.Path()
+		if err != nil {
+			return err
+		}
+		// read the old version file
+		versionPath := filepath.Join(repoPath, repo.LegacyVersionFile)
+		versionBytes, err := os.ReadFile(versionPath)
+		if err != nil {
+			return err
+		}
+		var version repo.Version
+		if err := json.Unmarshal(versionBytes, &version); err != nil {
+			return err
+		}
+
+		// write version 4 to the system_metadata.yaml file
+		if err := r.WriteVersion(repo.Version4); err != nil {
+			return err
+		}
+
+		_, fileCfg, err := readConfig(r)
+		if err != nil {
+			return err
+		}
+		if fileCfg.User.InstallationID != "" {
+			if err := r.WriteInstallationID(fileCfg.User.InstallationID); err != nil {
+				return err
+			}
+		}
```

Comment:
we are going to have a lot of fun with migrations when we change the config structure. Maybe we can keep a copy of the old config structs inside migrations pkg. Will need to figure this out later